camayoc/tests/qcs/api/v1/scanjobs/test_run_scanjobs.py

```diff
@@ -303,3 +314,76 @@ def test_scan_task_results(scan_info):
+            a) Create a facts_to_ignore list composed of raw_facts for each
+                disabled role
+            b) Iterate through the inspection results fact dictionaries
+            c) Assert that no facts that should be ignored are in the
+                dictionaries
+    :expectedresults: No facts are collected for disabled products
+    """
+    FOUND = False
+    scans = scan_info()
+    facts_to_ignore = []
+    for scan in scans:
+        disabled_optional_products = scan.get('disabled_optional_products')
+        if disabled_optional_products:
+            # Build the list of facts that should not be in inspection results
+            for product in disabled_optional_products.keys():
+                if product == 'jboss_eap':
+                    facts_to_ignore += QCS_EAP_RAW_FACTS
+                elif product == 'jboss_fuse':
+                    facts_to_ignore += QCS_FUSE_RAW_FACTS
+                elif product == 'jboss_brms':
+                    facts_to_ignore += QCS_BRMS_RAW_FACTS
+            # grab the inspection results of the scan
+            inspection_results = \
+                get_scan_result(scan.get('name')).get('inspection_results')
+            for system in inspection_results:
+                fact_dicts = system.get('facts')
+                # grab the facts for each system
+                for dictionary in fact_dicts:
+                    for fact in facts_to_ignore:
+                        if fact in dictionary.values():
+                            FOUND = True
+        assert not FOUND
+
+
+def test_disabled_optional_products():
+    """Test scan jobs from scans with disabled_optional_products.
+
+    :id: 77aa1f0c-32be-11e8-b467-0ed5f89f718b
+    :description: Test that scans created with disabled products
+        retain the same disabled products when results are obtained
+        from the api
+    :steps: 1) Iterate over scans and gather disabled products
+            2) If the scans have disabled products, grab the
+                disabled_products that were returned in the scan results
+            3) Check that each specified product has the same value
+                after being returned from the api
+    :expectedresults: The values for disabled products specified in the
+        config file are the same as those returned from the api
+    """
+    scans = scan_info()
+    for scan in scans:
+        # grab disabled products from config file
+        specified_optional_products = scan.get('disabled_optional_products')
+        if specified_optional_products:
+            # grab disabled products from results
+            returned_optional_products = \
+                get_scan_result(scan.get('name')).get('scan_results').get(
+                    'options').get('disabled_optional_products')
+            for product in specified_optional_products:
+                assert specified_optional_products[product] == \
```

Comment:
Just thought of one more thing. Asserting on each product inside the loop means the test stops after the first failure.

If you can instead collect the errors and after all products and all scans assert that there were no errors, and raise some collection of messages that are informative enough so we can know whats going on from the failure on Jenkins -- this is best.

```diff
    scans = scan_info()
+ errors_found = []
    for scan in scans:
        # grab disabled products from config file
        specified_optional_products = scan.get('disabled_optional_products')
        if specified_optional_products:
            # grab disabled products from results
            returned_optional_products = \
                get_scan_result(scan.get('name')).get('scan_results').get(
                    'options').get('disabled_optional_products')
           for product in specified_optional_products:
+                if specified_optional_products[product]  != returned_optional_products[product]:
+                    errors_found.append( 
+                                  'Found facts collected for {product_name} that should have been DISABLED'
+                                  ' on scan named {scan_name}'.format(product_name=product, scan_name=scan.get('name'))
+    assert len(errors_found) == 0, '\n================\n'.join(errors_found)
```
Something like that. Did not test this probably missing a quote or something like that.